File: IsuCorpReservation/Res.AspAngular/ClientApp/src/app/models/reserve.model.ts
import { CustomerViewModel } from "../models/customer.model";

export class ReserveViewModel{
    public id : number;
    public restaurant : string;
    public dateReserve : Date;
    public ranking : number;
    public favoriteStatus : boolean;
    public customerId : number;
    public customer : CustomerViewModel;

    constructor(date : Date, restaurant : string, custId : number)
    {
        this.dateReserve = (!date) ? new Date(): date;
        this.restaurant = restaurant;
        this.customerId = custId;
    }
}

File: IsuCorpReservation/Res.ApplicationLayer/Models/ReserveModel.cs
using Res.ApplicationLayer.Models.Base;
using System;

namespace Res.ApplicationLayer.Models
{
    public class ReserveModel : BaseModel
    {
        public string Restaurant { get; set; }

        public DateTime DateReserve { get; set; }

        public int Ranking { get; set; }

        public bool FavoriteStatus { get; set; }

        public int CustomerId { get; set; }

        public CustomerModel Customer { get; set; }
    }
}
